test/unit/test/pretty_printing_test.cpp

```diff
@@ -37,78 +34,68 @@ auto debug_str = [](auto && v)
+TEST(pretty_printing, vector)
 {
-    // if any of these tests get supported by googletest move them to gtest_output
     EXPECT_EQ(gtest_str(std::vector<std::vector<int>>{{0, 1}, {2, 3}, {1, 2}, {0}}), "[[0,1],[2,3],[1,2],[0]]"s);
     EXPECT_EQ(debug_str(std::vector<std::vector<int>>{{0, 1}, {2, 3}, {1, 2}, {0}}), "[[0,1],[2,3],[1,2],[0]]"s);
-
-    EXPECT_EQ(gtest_str(std::nullopt), "<VALUELESS_OPTIONAL>"s);
-    EXPECT_EQ(debug_str(std::nullopt), "<VALUELESS_OPTIONAL>"s);
 }
 
-TEST(pretty_printing, seqan3_output)
+TEST(pretty_printing, dna)
 {
-    // seqan3 types should always produce the same result
     EXPECT_EQ(gtest_str('G'_dna4), "G"s);
     EXPECT_EQ(debug_str('G'_dna4), "G"s);
-    EXPECT_EQ('G'_dna4, 'G'_dna4); // change value to test
+}
 
+TEST(pretty_printing, dna_sequence)
+{
     EXPECT_EQ(gtest_str("ACGTCGA"_dna4), "ACGTCGA"s);
     EXPECT_EQ(debug_str("ACGTCGA"_dna4), "ACGTCGA"s);
-    EXPECT_EQ("ACGTCGA"_dna4, "ACGTCGA"_dna4); // change value to test
-
-    std::vector dna_set1{"AC"_dna4, "GT"_dna4, "CG"_dna4, "A"_dna4};
-    std::vector dna_set2{"AC"_dna4, "GT"_dna4, "CG"_dna4, "A"_dna4};
-    EXPECT_EQ(gtest_str(dna_set1), "[AC,GT,CG,A]"s);
-    EXPECT_EQ(debug_str(dna_set1), "[AC,GT,CG,A]"s);
-    EXPECT_EQ(dna_set1, dna_set2); // change value to test
-
-    std::vector dna_sets1{std::vector{"AC"_dna4, "GT"_dna4}, std::vector{"CG"_dna4, "A"_dna4}};
-    std::vector dna_sets2{std::vector{"AC"_dna4, "GT"_dna4}, std::vector{"CG"_dna4, "A"_dna4}};
-    EXPECT_EQ(gtest_str(dna_sets1), "[[AC,GT],[CG,A]]"s);
-    EXPECT_EQ(debug_str(dna_sets1), "[[AC,GT],[CG,A]]"s);
-    EXPECT_EQ(dna_sets1, dna_sets2); // change value to test
+
+    std::vector dna_2d{"AC"_dna4, "GT"_dna4, "CG"_dna4, "A"_dna4};
+    EXPECT_EQ(gtest_str(dna_2d), "[AC,GT,CG,A]"s);
+    EXPECT_EQ(debug_str(dna_2d), "[AC,GT,CG,A]"s);
+
+    std::vector dna_3d{std::vector{"AC"_dna4, "GT"_dna4}, std::vector{"CG"_dna4, "A"_dna4}};
+    EXPECT_EQ(gtest_str(dna_3d), "[[AC,GT],[CG,A]]"s);
+    EXPECT_EQ(debug_str(dna_3d), "[[AC,GT],[CG,A]]"s);
 }
 
-TEST(pretty_printing, gtest_mixed_seqan3_output)
+TEST(pretty_printing, dna_tuple)
 {
-    // warning: these outputs can change due to changes in gtest
-    auto dna_tuple1 = std::make_tuple('A'_dna4, 'C'_dna4);
-    auto dna_tuple2 = std::make_tuple('A'_dna4, 'C'_dna4);
-    EXPECT_EQ(gtest_str(dna_tuple1), "(A, C)"s);
-    EXPECT_EQ(debug_str(dna_tuple1), "(A,C)"s);
-    EXPECT_EQ(dna_tuple1, dna_tuple2); // change value to test
+    auto dna_tuple = std::make_tuple('A'_dna4, 'C'_dna4);
+    EXPECT_EQ(gtest_str(dna_tuple), "(A, C)"s);
+    EXPECT_EQ(debug_str(dna_tuple), "(A,C)"s);
 
-    auto dna_sequence_tuple1 = std::make_tuple("AC"_dna4, "GT"_dna4);
-    auto dna_sequence_tuple2 = std::make_tuple("AC"_dna4, "GT"_dna4);
-    EXPECT_EQ(gtest_str(dna_sequence_tuple1), "(AC, GT)"s);
-    EXPECT_EQ(debug_str(dna_sequence_tuple1), "(AC,GT)"s);
-    EXPECT_EQ(dna_sequence_tuple1, dna_sequence_tuple2); // change value to test
```

Comment:
you always removed the "change value to test". Why? seemed helpful for testing, no?

Reply:
I think you could alter one of the tuples and than the EXPECT_EQ will fail which triggers the output of gtest so you can adapt the gtest output?

But now that you removed it it's probably not worth the hassle to reintroduce.

Reply:
Ah, that wasn't clear at all from the comment :D

But if gtest changes, the `gtest_str` also changes and prints the expected/actual gtest output

Reply:
yes propbably. You can leave the PR as is FMPOV